cmd/system-probe/api/server.go

```diff
@@ -48,8 +59,21 @@ func StartServer(cfg *config.Config, telemetry telemetry.Component) error {
 	mux.Handle("/debug/vars", http.DefaultServeMux)
 	mux.Handle("/telemetry", telemetry.Handler())
 
+	if cfg.GRPCServerEnabled {
```

Comment:
I don't understand the need for `RegisterGRPC`, or passing in the `grpcServer` to `Register`. Since its either going to be a grpc server or http, can't we just pass in just `mux` as before?

Reply:
we need to pass grpcServer to register the gRPC endpoints on the server. Mux is a HTTP router and does not support gRPC.

regardless, the gRPC servers are not replacing the http server, we only allow implementing different endpoints via HTTP or gRPC (up to the module to decide). In the future PR, we'll introduce a gRPC endpoint to get connections (similarly to the HTTP endpoint), but they meant to co-exist and not  to replace each other